ghmirror/core/mirror_requests.py

```diff
@@ -32,24 +32,56 @@
 LOG = logging.getLogger(__name__)
 
 
+def _get_elements_per_page(url_params):
+    """
+    Get 'per_page' parameter if present in URL or
+    return None if not present
+    """
+    if url_params.get('per_page'):
+        return int(url_params.get('per_page'))
+
+    return None
+
+
+def _cache_response(resp, cache, cache_key):
+    """
+    Implements the logic to decide whether or not
+    whe should cache a request acording to the headers
+    and content
+    """
+    # Caching only makes sense when at least one
+    # of those headers is present
+    if resp.status_code == 200 and \
+       any(['ETag' in resp.headers, 'Last-Modified' in resp.headers]):
+        cache[cache_key] = resp
+
+
 @requests_metrics
-def conditional_request(method, url, auth, data=None):
+def conditional_request(method, url, auth, data=None, url_params=None):
     """
     Implements conditional requests, checking first whether
     the upstream API is online of offline to decide which
     request routine to call.
     """
     if GithubStatus().online:
-        return online_request(method, url, auth, data)
+        return online_request(method, url, auth, data, url_params)
     return offline_request(method, url, auth)
 
 
-def online_request(method, url, auth, data=None):
+def online_request(method, url, auth, data=None, url_params=None):
     """
     Implements conditional requests.
     """
     cache = RequestsCache()
     headers = {}
+    parameters = url_params.to_dict() if url_params else {}
```

Comment:
For correctness: `parameters = url_params.to_dict() if url_params is not None else {}`